internal/node/manager_test.go

```diff
@@ -586,3 +759,253 @@ func TestContractDeploymentAndInteraction(t *testing.T) {
 	eventVal := new(big.Int).SetBytes(data)
 	require.Equal(t, int64(7), eventVal.Int64())
 }
+
+// TestPeerConnectivity_TwoNodes verifies basic peer-to-peer network formation between two nodes.
+//
+// Blockchains are peer-to-peer networks - nodes must connect to share blocks and transactions.
+// Without peer connections, each node is an isolated island, unable to participate in consensus.
+//
+// This test creates a minimal network:
+//  1. Starts two independent nodes
+//  2. Tells node1 about node2's address (like sharing a phone number)
+//  3. Verifies they establish connection
+//  4. Checks both report having at least one peer
+//
+// The connection uses "enode" URLs - unique addresses that identify nodes:
+//
+//	enode://[public-key]@[ip]:[port]
+//	Like: enode://abc123...@127.0.0.1:30303
+//
+// Why peer connectivity matters:
+//   - Blocks must propagate to all nodes for consensus
+//   - Transactions spread through peer gossip
+//   - Isolated nodes can't participate in the network
+//   - Network partition would create competing chains
+//
+// This is the foundation of decentralization - nodes discovering and maintaining
+// connections without central coordination.
+func TestPeerConnectivity_TwoNodes(t *testing.T) {
+	ctx, cancel, manager := startNodes(t, 2)
+	defer cancel()
+
+	require.Equal(t, 2, manager.NodeCount())
+
+	node1 := manager.GetNode(0)
+	node2 := manager.GetNode(1)
+	require.NotNil(t, node1)
+	require.NotNil(t, node2)
+
+	node2Enode := node2.Server().NodeInfo().Enode
+
+	client1, err := rpc.DialContext(ctx, utils.LocalAddress(manager.RPCPort()))
+	require.NoError(t, err)
+	defer client1.Close()
+
+	client2, err := rpc.DialContext(ctx, utils.LocalAddress(manager.GetRPCPort(1)))
+	require.NoError(t, err)
+	defer client2.Close()
+
+	// ensure node1 knows about node2
+	require.NoError(t, client1.CallContext(ctx, nil, "admin_addPeer", node2Enode))
+
+	// wait for peer connections
+	require.Eventually(
+		t, func() bool {
+			var count string
+			if err := client1.CallContext(ctx, &count, model.NetPeerCount); err != nil {
+				return false
+			}
+			return count != "0x0"
+		}, 10*time.Second, 500*time.Millisecond, "peers did not connect",
+	)
```

Comment:
This test only verifies that node1 has peers, but doesn't check node2, I'd suggest verifying bidirectional connectivity: 

```suggestion
	)
	// Verify node2 also sees node1 as a peer
  var count2 string
  require.NoError(t, client2.CallContext(ctx, &count2, model.NetPeerCount))
  require.NotEqual(t, "0x0", count2, "node2 should have at least one peer")
```

Reply:
https://github.com/thep2p/go-eth-localnet/pull/37/commits/6817655b997107f2148a8dd29b87964ca9ce014f